src/main/java/org/example/springbootintro/repository/book/BookRepositoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.example.springbootintro.repository.book;
+
+import java.util.List;
+import org.example.springbootintro.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
this is also redundant if you have only one constructor with all the necessary parameters
```suggestion
```